src/ResourceManager/DevSpaces/Commands.DevSpaces/Commands/RemoveAzureRmDevSpacesController.cs

```diff
@@ -0,0 +1,122 @@
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System;
+using System.Management.Automation;
+using Microsoft.Azure.Commands.DevSpaces.Models;
+using Microsoft.Azure.Commands.DevSpaces.Properties;
+using Microsoft.Azure.Commands.DevSpaces.Utils;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Management.DevSpaces;
+
+namespace Microsoft.Azure.Commands.DevSpaces.Commands
+{
+    [Cmdlet(VerbsCommon.Remove, DevSpacesControllerNoun, DefaultParameterSetName = DevSpacesControllerNameParameterSet, SupportsShouldProcess = true)]
+    [OutputType(typeof(bool))]
+    public class RemoveAzureRmDevSpacesController : DevSpacesCmdletBase
+    {
+        [Parameter(
+            Position = 0,
+            Mandatory = true,
+            ParameterSetName = DevSpacesControllerNameParameterSet,
+            HelpMessage = "Resource group name")]
+        [ResourceGroupCompleter()]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(
+            Position = 1,
+            Mandatory = true,
+            ParameterSetName = DevSpacesControllerNameParameterSet,
+            HelpMessage = "DevSpaces controller name.")]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = ResourceIdParameterSet,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The DevSpaces resource id")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
+
+        [Parameter(
+            ParameterSetName = InputObjectParameterSet,
+            Mandatory = true,
+            ValueFromPipeline = true)]
+        [ValidateNotNullOrEmpty]
+        public PSController InputObject { get; set; }
+
+        [Parameter(Mandatory = false)]
+        public SwitchParameter PassThru { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "Run cmdlet in the background")]
+        public SwitchParameter AsJob { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            base.ExecuteCmdlet();
+            var msg = $"{Name} in {ResourceGroupName}";
+
+            if (ShouldProcess(msg, Resources.DeletingADevSpacesController))
+            {
+                RunCmdLet(RemoveDevSpacesControllerAction);
+            }
+        }
+
+        private void RemoveDevSpacesControllerAction()
+        {
+            switch (ParameterSetName)
+            {
+                case DevSpacesControllerNameParameterSet:
+                    break;
+
+                case ResourceIdParameterSet:
+                    string resourceGroup, name;
+                    if (!ConversionUtils.TryParseResourceId(ResourceId, ConversionUtils.DevSpacesControllerResourceTypeName, out resourceGroup, out name))
```

Comment:
Same comment about ResourceIdentifier.

Reply:
ConversionUtils.TryParseResourceId is implemented on top of ResourceIdentifier to make sure that we are parsing DevSpace or AKS ID.